controllers/vspherevm_controller.go

```diff
@@ -148,20 +143,22 @@ func AddVMControllerToManager(ctx context.Context, controllerManagerCtx *capvcon
 }
 
 type vmReconciler struct {
-	*capvcontext.ControllerContext
-
+	record.Recorder
+	*capvcontext.ControllerManagerContext
 	VMService                 services.VirtualMachineService
 	remoteClusterCacheTracker *remote.ClusterCacheTracker
 }
 
 // Reconcile ensures the back-end state reflects the Kubernetes resource state intent.
 func (r vmReconciler) Reconcile(ctx context.Context, req ctrl.Request) (_ ctrl.Result, reterr error) {
-	r.Logger.V(4).Info("Starting Reconcile", "key", req.NamespacedName)
+	logger := ctrl.LoggerFrom(ctx)
```

Comment:
```suggestion
	log := ctrl.LoggerFrom(ctx)
```

Nit - seems to be the more common pattern, but we have both.

Reply:
done - along with other instances found across files, in-scope of this MR.